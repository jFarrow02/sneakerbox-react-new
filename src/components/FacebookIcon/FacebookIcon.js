import React from 'react';
import './FacebookIcon.scss';
import {useFill} from '../../hooks/uiHooks';

const FacebookIcon = props=>{
    
    //Re-use 'stateful logic' from custom hook:
    const [fill, setFill] = useFill(props.initialFill);

    return(
        <svg
            alt="facebook"
            className='facebook-logo'
            focusable='false'
            height={props.height}
            onMouseEnter={()=>{setFill(props.newFill)}}
            onMouseLeave={()=>{setFill(props.initialFill)}}
            viewBox="0 0 512 512" 
            width={props.width}
            xmlns="http://www.w3.org/2000/svg"
        >
            <path 
                d="m437 0h-362c-41.351562 0-75 33.648438-75 75v362c0 41.351562 33.648438 75 75 75h151v-181h-60v-90h60v-61c0-49.628906 40.371094-90 90-90h91v90h-91v61h91l-15 90h-76v181h121c41.351562 0 75-33.648438 75-75v-362c0-41.351562-33.648438-75-75-75zm0 0"
                fill={fill}
            />
        </svg>
    )
}

export default FacebookIcon;