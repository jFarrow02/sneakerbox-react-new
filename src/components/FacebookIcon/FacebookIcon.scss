@import '../../styles/variables.scss';

.facebook-logo:hover{
    fill: $copper;
}
