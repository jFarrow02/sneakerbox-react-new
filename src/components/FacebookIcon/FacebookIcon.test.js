import React from 'react';
import FacebookIcon from './FacebookIcon';
import renderer from 'react-test-renderer';

it('should match a snapshot', ()=>{
    const tree = renderer.create(<FacebookIcon/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
})