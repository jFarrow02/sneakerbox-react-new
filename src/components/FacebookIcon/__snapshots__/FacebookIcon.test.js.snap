// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match a snapshot 1`] = `
<svg
  alt="facebook"
  className="facebook-logo"
  focusable="false"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="m437 0h-362c-41.351562 0-75 33.648438-75 75v362c0 41.351562 33.648438 75 75 75h151v-181h-60v-90h60v-61c0-49.628906 40.371094-90 90-90h91v90h-91v61h91l-15 90h-76v181h121c41.351562 0 75-33.648438 75-75v-362c0-41.351562-33.648438-75-75-75zm0 0"
    fill="#fff9d4"
  />
</svg>
`;
